### PR TITLE
Add step indicator

### PR DESCRIPTION
**Pull request by Rostislav013**
image

image

image

### CODE REVIEW

**Comment by fshauge** [comment]
It looks like, according to sketches in Figma, the gray steps shouldn't show the step number

File: src/Nordea/Components/StepIndicator.elm [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+view : StepIndicator -> Html msg
+view (StepIndicator config) =
+    div []
+        [ viewSteps config.stepNames config.currentStep
+        ]
+
+
+enumerateList : List a -> List ( Int, a )
+enumerateList list =
+    List.zip (List.range 1 (List.length list)) list
```
Comment (kseverinsen) [inline src/Nordea/Components/StepIndicator.elm:75]:
```suggestion
    List.indexedMap Tuple.pair
```

File: src/Nordea/Components/StepIndicator.elm [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            ]
+
+    else
+        styled li
+            listStylesLi
+            []
+            [ styled div divStyles [] [ text (String.fromInt item) ]
+            , styled p textStyles [] [ text stepName ]
+            , viewLine
+            ]
```
Comment (kseverinsen) [inline src/Nordea/Components/StepIndicator.elm:131]: The first `if` case could be eliminated by conditionally showing the line element
```elm
showIf : Bool -> Html msg -> Html msg
showIf pred elem =
    if pred then
        elem

    else
        text ""
```

```suggestion
viewStepItem : Int -> ( Int, String ) -> Html msg
viewStepItem currentStep ( item, stepName ) =
    if item < currentStep then
        styled li
            listStylesLi
            []
            [ styled div divStylesActive [] [ Icons.check ]
            , styled p textStyles [] [ text stepName ]
            , showIf (item /= 1) viewLineActive
            ]

    else if item == currentStep then
        styled li
            listStylesLi
            []
            [ styled div divStylesCurrent [] [ text (String.fromInt item) ]
            , styled p textStyles [] [ text stepName ]
            , showIf (item /= 1) viewLineActive
            ]

    else
        styled li
            listStylesLi
            []
            [ styled div divStyles [] [ text (String.fromInt item) ]
            , styled p textStyles [] [ text stepName ]
            , showIf (item /= 1) viewLine
            ]
```

**Comment by Rostislav013** [comment]
> It looks like, according to sketches in Figma, the gray steps shouldn't show the step number

I have talked to Beate last time, she said it's ok with numbers in gray steps too

**Pull request merged**